enhancements/network/user-defined-network-segmentation.md

```diff
@@ -379,11 +382,116 @@ any user-defined primary network.
+without the risk of destabilizing the cluster nodes or break the cluster network.
+
+There should be a finalizer on the CRDs, so that upon deletion OVN-Kubernetes can validate that there are no pods still using this network.
+If there are pods still attached to this network, the network will not be removed.
+
+##### spec
+
+The CRDs spec defines as follows:
+
+| Field name     | Description                                                                                                                                                                                                                                                                                                                                                                                        | optional |
+|----------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------|
+| Topology       | The topological configuration for the network. Must be one of `Layer2`, `layer3`.                                                                                                                                                                                                                                                                                                                  | No       |
+| Role           | Select the network role in the pod, either `Primary` or `Secondary`.                                                                                                                                                                                                                                                                                                                               | No       |
+| MTU            | The maximum transmission unit (MTU).<br/>The default is 1400.                                                                                                                                                                                                                                                                                                                                      | Yes      |
+| Subnets        | The subnet to use for the network across the cluster.<br/>E.g. 10.100.200.0/24.<br/>IPv6 (2001:DBB::/64) and dual-stack (192.168.100.0/24,2001:DBB::/64) subnets are supported.<br/>When omitted, the logical switch implementing the network only provides layer 2 communication, and users must configure IP addresses.<br/>Port security only prevents MAC spoofing if the subnets are omitted. | Yes      |
```

Comment:
Additionally join subnet or regular subnet cannot be modified after CR creation. You need to add this to the table, and then in the CRD we need CEL to enforce this, like:

```
                 
                  x-kubernetes-validations:
                  - rule: "self == oldSelf"
                    message: "The field 'subnet' cannot be updated once it is set."
```

Reply:
The whole CRDs spec should be immutable, isn't it?

Since we utilize NADs, mutating the request NAD may lead to incomplete state where some pod configured by a NAD before spec is changed (gen 1) and new pods will be configured by a NAD with new spec (gen2).

Reply:
hmm good point. I guess for now the best thing to do is make it immutable. In the future we may want to open the door for certain fields to be changed. Will think a little bit more about this.